pkg/ccl/changefeedccl/sink_webhook.go

```diff
@@ -571,18 +578,30 @@ func (s *webhookSink) workerLoop(workerIndex int) {
 				continue
 			}
 
-			encoded, err := encodePayloadWebhook(msgs)
-			if err != nil {
-				s.exitWorkersWithError(err)
-				return
-			}
-			if err := s.sendMessageWithRetries(s.workerCtx, encoded.data); err != nil {
-				s.exitWorkersWithError(err)
-				return
+			switch s.format {
+			case changefeedbase.OptFormatJSON:
+				encoded, err := encodePayloadWebhook(msgs)
+				if err != nil {
+					s.exitWorkersWithError(err)
+					return
+				}
+				if err := s.sendMessageWithRetries(s.workerCtx, encoded.data); err != nil {
+					s.exitWorkersWithError(err)
+					return
+				}
+				encoded.alloc.Release(s.workerCtx)
+				s.metrics.recordEmittedBatch(
+					encoded.emitTime, len(msgs), encoded.mvcc, len(encoded.data), sinkDoesNotCompress)
+			case changefeedbase.OptFormatCSV:
```

Comment:
can we change encodePayloadWebhook to handle both formats? This way, we won't have to duplicate sendmessagewith retries.. and also, forget to record metrics?

Alternatively:
```
  batchEncoder := func() func(messages []messagePayload) (encodedPayload, error)  {
      if JSON {
         return encodePayloadJson
     }
     return encodePayloadCSV
}()
```
Then just use batchEncoder() instead of calling encodePayloadWebhook.

Reply:
Thanks for the suggestion! I completely agree 🙌🏽